### DISCUSSION TITLE
useSelector question

### DISCUSSION THREAD

**Discussion by AleksandrukTad** [pmndrs/jotai#1785] [Q&A] [answered]
Hey, 
I have found this post: https://github.com/pmndrs/jotai/issues/36 I got question about it. Am I understanding this right that useSelector will slice the **bigAtom**, get a read only value from it and will not rerender the component when some other value in **bigAtom** is updated?

example: 
const errorsAtoms = atom({ firstName: "", secondName: "", lastName: "" })

const firstNameError = useSelector(errorsAtoms , (errors) => errors.firstName);
const secondNameError = useSelector(errorsAtoms , (errors) => errors.secondName);
const lastNameError = useSelector(errorsAtoms , (errors) => errors.lastName);

When I have tree like that

Form
  - FirstName
  - SecondName
  - LastName
  
When the the **secondName** fails validation during submit and now errorsAtoms is like so: 
const errorsAtoms = atom({ firstName: "", secondName: "Failed validation", lastName: "" })

Will it rerender FirstName component?

**Comment by dai-shi** [comment, marked as answer]
> Will it rerender FirstName component?

No. I think it work as you expect.

useSelector is a bit old, current recommendation would be simply deriving atoms:
```js
const errorsAtom = atom({ firstName: "", secondName: "", lastName: "" })

const firstNameErrorAtom = atom((get) => get(errorsAtom).firstName);
const secondNameErrorAtom = atom((get) => get(errorsAtom).secondName);
const lastNameErrorAtom = atom((get) => get(errorsAtom).lastName);
```

**Reply by tjx666** [reply]
Why `deriving atoms` is the recommend way? I think `useSelector` is convenient enough and good for performance.

**Reply by dai-shi** [reply]
Because we don't need to care about making selectors stable.

I don't think there's a difference on performance. It's fine to have `useSelector` on userland. There are several ways to implement it depending on requirements.
